crates/edr_napi/src/trace/error_inferrer.rs

```diff
@@ -0,0 +1,2426 @@
+                type_: StackTraceEntryTypeConst,
+                source_reference,
+                value: trace.value.clone(),
+            }
+            .into()]));
+        }
+
+        Ok(None)
+    }
+
+    pub fn infer_before_tracing_create_message(
+        trace: &CreateMessageTrace,
+        env: Env,
+    ) -> napi::Result<Option<SolidityStackTrace>> {
+        if Self::is_constructor_not_payable_error(trace, env)? {
+            return Ok(Some(vec![FunctionNotPayableErrorStackTraceEntry {
+                type_: StackTraceEntryTypeConst,
+                source_reference: Self::get_constructor_start_source_reference(trace, env)?,
+                value: trace.value.clone(),
+            }
+            .into()]));
+        }
+
+        if Self::is_constructor_invalid_arguments_error(trace, env)? {
+            return Ok(Some(vec![InvalidParamsErrorStackTraceEntry {
+                type_: StackTraceEntryTypeConst,
+                source_reference: Self::get_constructor_start_source_reference(trace, env)?,
+            }
+            .into()]));
+        }
+
+        Ok(None)
+    }
+
+    pub fn infer_after_tracing(
+        trace: Either<&CallMessageTrace, &CreateMessageTrace>,
+        stacktrace: SolidityStackTrace,
+        function_jumpdests: &[&Instruction],
+        jumped_into_function: bool,
+        last_submessage_data: Option<SubmessageDataRef<'_>>,
+        env: Env,
+    ) -> napi::Result<SolidityStackTrace> {
+        /// Convenience macro to early return the result if a heuristic hits.
+        macro_rules! return_if_hit {
+            ($heuristic: expr) => {
+                match $heuristic {
+                    Heuristic::Hit(stacktrace) => return Ok(stacktrace),
+                    Heuristic::Miss(stacktrace) => stacktrace,
+                }
+            };
+        }
+
+        let result = Self::check_last_submessage(trace, stacktrace, last_submessage_data, env)?;
+        let stacktrace = return_if_hit!(result);
+
+        let result = Self::check_failed_last_call(trace, stacktrace, env)?;
+        let stacktrace = return_if_hit!(result);
+
+        let result = Self::check_last_instruction(
+            trace,
+            stacktrace,
+            function_jumpdests,
+            jumped_into_function,
+            env,
+        )?;
+        let stacktrace = return_if_hit!(result);
+
+        let result = Self::check_non_contract_called(trace, stacktrace, env)?;
+        let stacktrace = return_if_hit!(result);
+
+        let result = Self::check_solidity_0_6_3_unmapped_revert(trace, stacktrace, env)?;
+        let stacktrace = return_if_hit!(result);
+
+        if let Some(result) = Self::check_contract_too_large(trace, env)? {
+            return Ok(result);
+        }
+
+        let stacktrace = Self::other_execution_error_stacktrace(trace, stacktrace, env)?;
+        Ok(stacktrace)
+    }
+
+    pub fn filter_redundant_frames(
+        stacktrace: SolidityStackTrace,
```

Comment:
That is true but I only changed the PascalCase to snake_case but didn't modify the word boundaries to make the review easier to follow and because some N-API exported types used this wording (they do not now, anymore).

Would you like me to change every occurrence of this, here?

Reply:
I saw StackTrace used in some of the types, so I thought this might have been erroneous.

You can fix it now or in a follow-up PR.

Reply:
The original code might have (incorrectly) had `stacktrace` as a single word in some places. Agree that it should be two words. I think it's better to do that in a separate PR though.

Reply:
Let's do that in a separate PR, then.